### PR TITLE
Improve python3 and Django 1.11 and 2.x compatibility

### PR DESCRIPTION
**Pull request by mmuzila**
Currently kobo is not able to work with Django 2.x flawlessly. Improve Django 1.11 and 2.x compatibility, preserving  backwards compatibility with Django 1.7.

### CODE REVIEW

File: kobo/hub/migrations/0001_initial.py [hunk 1]
```diff
@@ -51,8 +51,8 @@ class Migration(migrations.Migration):
                 ('priority', models.PositiveIntegerField(default=10, help_text='Priority.')),
                 ('weight', models.PositiveIntegerField(default=1, help_text='Weight determines how many resources is used when processing the task.')),
                 ('subtask_count', models.PositiveIntegerField(default=0, help_text='Subtask count.<br />This is a generated field.')),
-                ('arch', models.ForeignKey(to='hub.Arch')),
-                ('channel', models.ForeignKey(to='hub.Channel')),
+                ('arch', models.ForeignKey(to='hub.Arch', on_delete=models.CASCADE)),
```
Comment (yontalcar) [inline kobo/hub/migrations/0001_initial.py]: IIUIC this should go into new migration file. Same for other changes in `*/migrations/*`

File: kobo/hub/urls/task.py [hunk 2]
```diff
@@ -1,17 +1,18 @@
 # -*- coding: utf-8 -*-
 
 
-from django.utils.translation import ugettext_lazy as _
+from dango.utils.translation import ugettext_lazy as _
```
Comment (querti) [inline kobo/hub/urls/task.py]: typo

**Comment by rohanpm** [comment]
@mmuzila, are you willing to look into the test failures on this PR? (can be seen in Travis logs)

It can't be submitted as long as there are failures.

**Comment by rohanpm** [comment]
I'm closing this, as most (or all?) of the commits here ended up being incorporated into https://github.com/release-engineering/kobo/pull/134, which was ultimately merged.  The work is available in kobo 0.13.0 now.

**Pull request closed without merging**